scripts/tuple_tree_generator/tuple_tree_generator/templates/struct.h.tpl

```diff
@@ -142,9 +155,33 @@ public:
   bool operator>(const /*= struct.name =*/ &Other) const { return key() > Other.key(); }
 
   /** else **/
-  bool operator==(const /*= struct.name =*/ &Other) const = default;
+  bool operator==(const /*= struct.name =*/ &Other) const {
+    /**- for field in struct.fields **/
+    if (/*= field.name =*/() != Other./*= field.name =*/())
+      return false;
+    /**- endfor **/
+    return true;
+  }
   /** endif **/
 
+  /**- if emit_tracking **/
+  llvm::SmallVector<revng::AccessTracker*, 3> getTrackers() const {
+    return {
+    /**- for field in struct.fields **/
+      &/*= field.name =*/Tracker,
+    /**- endfor **/
+    };
+  }
+
+  llvm::SmallVector<revng::AccessTracker*, 3> getKeyTrackers() const {
+    return {
+      /**- for key_field in struct.key_fields -**/
+      &/*= key_field.name =*/Tracker,
+      /**- endfor -**/
+    };
+  }
+  /** endif -**/
```

Comment:
1. You can return a `std::array`
2. Return `const revng::AccessTracker *`
3. Add space before `*`
4. Are you sure you want raw access to these? Maybe we should have these return a set of indexes representing the filed IDs that are changed and keep the fact we're using `AccessTracker` an implementation detail.

Reply:
4. We need to provide access to them since the visits are in cpp code. We can return the id the field i guess, but i don't see much reason to generate that code here rather than the cpp. Nobody but the visits has a interest in reading this stuff.

In particular we need at the very least the getTracker<Index> method, that method is outside of the class itself, and thus requires that the tracker itself is accessible, for that reason i made the public. I can write a generate for each toh. This accessors that return the list of them are just here for simplicity.

not sure which design you like most, but i really doubt we will encounter issue in the future regardless of which path we take.

Reply:
> We can return the id the field i guess, but i don't see much reason to generate that code here rather than the cpp

I'd be much happier to hide the implementation detail and make things private.
Let's return a set of indices. If you don't like generating too much code here, write an helper function in a header and use it here, but I don't see a reason to make any of this public.